src/spec.rs

```diff
@@ -1,43 +1,206 @@
+#![allow(clippy::wrong_self_convention)]
```

Comment:
Otherwise, we get this error (from the `from_le()`/`to_le()` methods) from clippy:
```rust
warning: methods called `from_*` usually take no `self`
   --> src/spec.rs:400:16
    |
400 |     fn from_le(mut self) -> Self {
    |                ^^^^^^^^
    |
    = help: consider choosing a less ambiguous name
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
```

Since these are private methods, I think this shouldn't matter. However, I agree that it should probably be applied to individual structs instead of the whole file. I'll make that change now!